flink-tests/src/test/java/org/apache/flink/test/streaming/api/FileReadingWatermarkITCase.java

```diff
@@ -62,8 +64,17 @@ public void testWatermarkEmissionWithChaining() throws Exception {
         JobExecutionResult result = env.execute();
 
         int actual = result.getAccumulatorResult(NUM_WATERMARKS_ACC_NAME);
-
-        assertTrue("too few watermarks emitted: " + actual, actual >= MIN_EXPECTED_WATERMARKS);
+        long expected =
+                result.<Long>getAccumulatorResult(RUNTIME_ACC_NAME) / WATERMARK_INTERVAL_MILLIS;
+        checkState(expected > MIN_EXPECTED_WATERMARKS);
+
+        assertEquals(
+                "too few watermarks emitted in "
+                        + result.<Long>getAccumulatorResult(RUNTIME_ACC_NAME)
+                        + " ms",
+                expected,
+                actual,
+                .5 * expected);
```

Comment:
Where does the delta come from? Isn't it too big? Does it mean that the number of checkpoints can be just half of the `expected` number?

Reply:
> Does it mean that the number of checkpoints can be just half of the expected number?

You mean watermarks, not checkpoints, right?

The value is just an empirical number because there are no guarantees regarding the watermark interval (or am I wrong?)
I'm afraid decreasing it would make the test more flaky (it does fail with `.2` sometimes and rarely with `.3`).

Reply:
Yes, I meant watermarks, sorry.

Reply:
Yeah, that's the sentiment that I am getting from the test. We just hope it works. ;)

Would simplifying the test to check if there was at least a single watermark made the test useless? Or would it basically check the same thing? I also don't get the comment on the class:

>  * Tests that watermarks are emitted while file is being read, particularly the last split.

How does it test emitting watermarks from the last split?

Reply:
Would be happy to learn how to test it better :)

Yes, I think single watermark will make it useless.

The comment is referring to the original issue, I'm not sure if it's still relevant.

Reply:
Just to be clear, I am not criticizing the work here. Just wanted to say it is quite a bit undeterministic.

Unfortunately I don't get the full purpose and corner cases it tries to test. I also don't understand where do the minimum 5 expected watermarks come from. Also don't know why we use 1 hour for bounded out of order watermarks, what's the criteria for the number of lines etc. 

I'll trust your judgement though. The changes definitely do not make it worse and if you say it will improve the stability let's go with them.

Reply:
I totally agree, it's undeterministic.
The values are empirical as there are no hard guarantees, e.g. with 5M lines watermark interval becomes stable.
1h comes from the originally reported case IIRC.